zipkin-cassandra/src/test/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStoreSpec.scala

```diff
@@ -36,4 +40,23 @@ class CassandraSpanStoreSpec extends SpanStoreSpec {
   override lazy val store = new CassandraSpanStore(new Repository(keyspace, cluster))
 
   override def clear = cluster.connect().execute("DROP KEYSPACE IF EXISTS " + keyspace)
+
+  override def setTimeToLive() {
+    ready(store(Seq(span1)))
+    ready(store.setTimeToLive(span1.traceId, 1234.seconds))
+
+    for( i <- 1 to 100) {
+      // @xxx Repository.storeXXX() methods write asynchronously but don't return a Future so we can't reliably test.
```

Comment:
It's not as bad as it sounds because it only is a problem in the context of tests.

The cassandra backend is intentionally eventually consistent here. Writes are written with only a consistency level of one, and so are reads. So if you had increased the replication of the cluster for the zipkin keyspace (something you definitely should do in production) then it's quite likely that there's a few microseconds after a write that if you were to read from a different node you want get the latest value.

And I can't see any reason in the zipkin datamodel that such truth, or strict consistency is ever required. Therefore the fire-and-forget of the writes is a triviality.

Except for the tests. But it's solved here in this pull request.